demisto_sdk/commands/validate/tests/BA_validators_test.py

```diff
@@ -2503,3 +2506,224 @@ def test_is_command_or_script_name_starts_with_digit_valid():
     )
 
     assert len(results) == 0
+
+
+@pytest.mark.parametrize(
+    "content_item, policy_names, expected_failures",
+    [
+        pytest.param(
+            create_script_object(
+                paths=["compliantpolicies"], values=[["valid_policy_1"]]
+            ),
+            {"valid_policy_1"},
+            0,
+            id="valid_policy_name",
+        ),
+        pytest.param(
+            create_script_object(
+                paths=["compliantpolicies"], values=[["invalid_policy"]]
+            ),
+            {"valid_policy_1", "valid_policy_2"},
+            1,
+            id="invalid_policy_name",
+        ),
+        pytest.param(
+            create_script_object(
+                paths=["compliantpolicies"],
+                values=[["valid_policy_1", "valid_policy_2"]],
+            ),
+            {"valid_policy_1", "valid_policy_2"},
+            0,
+            id="multiple_valid_policies",
+        ),
+        pytest.param(
+            create_script_object(
+                paths=["compliantpolicies"],
+                values=[["valid_policy_1", "invalid_policy"]],
+            ),
+            {"valid_policy_1", "valid_policy_2"},
+            1,
+            id="mixed_valid_and_invalid_policies",
+        ),
+        pytest.param(
+            create_script_object(paths=["compliantpolicies"], values=[[]]),
+            {"valid_policy_1"},
+            0,
+            id="empty_policy_list",
+        ),
+        pytest.param(
+            create_script_object(
+                paths=["compliantpolicies"],
+                values=[["invalid_policy_1", "invalid_policy_2"]],
+            ),
+            {"valid_policy_1", "valid_policy_2"},
+            2,
+            id="multiple_invalid_policies",
+        ),
+    ],
+)
+def test_script_compliant_policy_name_validator(
+    content_item, policy_names, expected_failures
+):
+    """
+    Given
+    - A Script object with compliant policies.
+    When
+    - Calling the IsValidCompliantPolicyNameValidator content_contains_invalid_compliant_policy_name function.
+    Then
+    - Make sure the expected number of failures are returned.
+
+    Test cases:
+    - A script with valid policy name
+    - A script with invalid policy name
+    - A script with multiple valid policies
+    - A script with a mix of valid and invalid policies
+    - A script with empty policy list
+    - A script with multiple invalid policies
+    """
+    results = IsValidCompliantPolicyNameValidator().content_contains_invalid_compliant_policy_name(
+        content_item, policy_names
+    )
+    assert len(results) == expected_failures
```

Comment:
Lets consider checking more then length.
```suggestion
    assert sorted(results) == sorted(expected_invalid_policies)
```